org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/ClassFile.java

```diff
@@ -4578,9 +4579,9 @@ private int generateRuntimeAnnotations(final Annotation[] annotations, final lon
 			if (annotationMask != 0 && (annotationMask & targetMask) == 0) {
 				if (!jdk16packageInfoAnnotation(annotationMask, targetMask)) continue;
 			}
```

Comment:
Is this jdk16packageInfoAnnotation stuff still relevant ??

Reply:
very good question :)  I had no idea about this, but it leads us back to https://bugs.eclipse.org/bugs/show_bug.cgi?id=449330 where Shankha wrote:

> If a annotation without the annotation target being set to PACKAGE is used on package-info.java we should allow that in 1.6 and below and generate the annotation in the class file based on the runtime policy. 

So, apart from the question whether supporting 1.6 is still relevant (unfortunately I lost any reference to where we discussed what can/should be dropped - it was mentioned in the not so distant past), perhaps the new JLS rule implemented here supersedes that special casing for 1.6-. 

But no, `jdk16packageInfoAnnotation` seems to admit annotations on package-info.java if they do have a `@Target` meta annotation but no mention of `PACKAGE` in it. That's different from what we find in current JLS.

Although the compiler regression suite run at -Dcompliance=1.6 does not need this code, I'm reluctant to remove it. Feel free to file a follow-up for clean-up.